tower-http/src/lib.rs

```diff
@@ -1,3 +1,83 @@
+//! `async fn(HttpRequest) -> Result<HttpResponse, Error>`
+//!
+//! # Overview
+//!
+//! tower-http is a library that provides HTTP specific middlewares and utilities built on top of
+//! [tower].
+//!
+//! All middlewares uses the [http] and [http-body] crates as the HTTP abstractions. That means
```

Comment:
nit: in the `tower` docs, at least, we use "middleware" as the plural, instead of "middlewares". i think this should be 
```suggestion
//! All middleware uses the [http] and [http-body] crates as the HTTP abstractions. That means
```